doozerlib/distgit.py

```diff
@@ -941,20 +902,32 @@ def _build_container(
             "--nowait",
         )
 
-        if odcs:
-            if odcs == 'signed':
-                odcs = 'release'  # convenience option for those used to the old types
-            cmd_list.append('--signing-intent')
-            cmd_list.append(odcs)
+        # Determine if ODCS is enabled by looking at container.yaml.
+        odcs_enabled = False
+        osbs_image_config_path = os.path.join(self.distgit_dir, "container.yaml")
+        if os.path.isfile(osbs_image_config_path):
+            with open(osbs_image_config_path, "r") as f:
+                image_config = yaml.safe_load(f)
+            odcs_enabled = "compose" in image_config
+
+        if odcs_enabled:
+            self.logger.info("About to build image in ODCS mode.")
+            if repo_type == 'signed':
+                signing_intent = 'release'
+            else:
+                signing_intent = repo_type
+            if signing_intent:
+                cmd_list.append('--signing-intent')
+                cmd_list.append(signing_intent)
         else:
             if repo_type:
                 repo_list = list(repo_list)  # In case we get a tuple
                 repo_list.append(self.metadata.cgit_url(".oit/" + repo_type + ".repo"))
 
-            if repo_list:
-                # rhpkg supports --repo-url [URL [URL ...]]
-                cmd_list.append("--repo-url")
-                cmd_list.extend(repo_list)
+        if repo_list:  # Current OSBS versions support the combination of ODCS composes with repository files.
```

Comment:
Was not the case before, but I see in https://mojo.redhat.com/docs/DOC-1159997#jive_content_id_Can_I_use_repourls_parameter_with_ODCS_composes that's changed...

However I don't think we want to take advantage of this. Everything we're shipping should be in pulp repos, except our own candidate tag, and for that we no longer want to point at "latest" -- that's kind of the whole point here, to use an ODCS compose for that.

Reply:
Do you think it is better to remove this feature when ODCS mode is enabled?

Reply:
I don't know if we ever specify a `--repo-list` to `doozer images:build` during runtime (looks like at lease we don't use that option in `aos-cd-jobs`). I reverted this part of change. If some day we do have the requirement of specifying runtime repo url even when ODCS is enabled, we can add it back.